primitives/src/addr.rs

```diff
@@ -610,114 +610,114 @@ mod std_compat {
         assert_eq!(
-            Ipv4Addr::from_str("0.0.0.0").unwrap(),
+            "0.0.0.0".parse::<Ipv4Addr>().unwrap(),
             Ipv4Addr([0, 0, 0, 0])
         );
 
         assert_eq!(
-            Ipv4Addr::from_str("127.0.0.1").unwrap(),
+            "127.0.0.1".parse::<Ipv4Addr>().unwrap(),
             Ipv4Addr([127, 0, 0, 1])
         );
 
         assert_eq!(
-            Ipv4Addr::from_str("192.168.1.256").unwrap_err(),
+            "192.168.1.256".parse::<Ipv4Addr>().unwrap_err(),
             ParseError::InvalidSegment
         );
 
         assert_eq!(
-            Ipv4Addr::from_str("192.168.1").unwrap_err(),
+            "192.168.1".parse::<Ipv4Addr>().unwrap_err(),
             ParseError::NotEnoughSegments
         );
 
         assert_eq!(
-            Ipv4Addr::from_str("192.168.1.2.3").unwrap_err(),
+            "192.168.1.2.3".parse::<Ipv4Addr>().unwrap_err(),
             ParseError::TooManySegments
         );
     }
 
     #[test]
     fn ipv6() {
         assert_eq!(
-            Ipv6Addr::from_str("::1").unwrap(),
+            "::1".parse::<Ipv6Addr>().unwrap(),
             Ipv6Addr([0, 0, 0, 0, 0, 0, 0, 1])
         );
 
         assert_eq!(
-            Ipv6Addr::from_str("ff02::1").unwrap(),
+            "ff02::1".parse::<Ipv6Addr>().unwrap(),
             Ipv6Addr([0xff02, 0, 0, 0, 0, 0, 0, 1])
         );
 
         assert_eq!(
-            Ipv6Addr::from_str("2001:0db8::").unwrap(),
+            "2001:0db8::".parse::<Ipv6Addr>().unwrap(),
             Ipv6Addr([0x2001, 0xdb8, 0, 0, 0, 0, 0, 0])
         );
 
         assert_eq!(
-            Ipv6Addr::from_str("2001:0db8:0000:0000:0000:0000:0000:0001").unwrap(),
+            "2001:0db8:0000:0000:0000:0000:0000:0001"
+                .parse::<Ipv6Addr>()
+                .unwrap(),
             Ipv6Addr([0x2001, 0xdb8, 0, 0, 0, 0, 0, 1])
         );
 
         assert_eq!(
-            Ipv6Addr::from_str("2001:0db8::0001").unwrap(),
+            "2001:0db8::0001".parse::<Ipv6Addr>().unwrap(),
             Ipv6Addr([0x2001, 0xdb8, 0, 0, 0, 0, 0, 1])
         );
 
         assert_eq!(
-            Ipv6Addr::from_str("2001:db8:0:1:2:3:4").unwrap_err(),
+            "2001:db8:0:1:2:3:4".parse::<Ipv6Addr>().unwrap_err(),
             ParseError::NotEnoughSegments
         );
 
         assert_eq!(
-            Ipv6Addr::from_str("2001:db8:0:1:2:3:4:5:6").unwrap_err(),
+            "2001:db8:0:1:2:3:4:5:6".parse::<Ipv6Addr>().unwrap_err(),
             ParseError::TooManySegments
         );
     }
 
     #[test]
     fn socket_v4() {
         assert_eq!(
-            SocketAddrV4::from_str("192.168.1.0:9019").unwrap(),
+            "192.168.1.0:9019".parse::<SocketAddrV4>().unwrap(),
             SocketAddrV4 {
                 ip: Ipv4Addr([192, 168, 1, 0]),
                 port: 9019
             }
         );
 
         assert_eq!(
-            SocketAddrV4::from_str("192.168.1.1").unwrap_err(),
+            "192.168.1.1".parse::<Ipv4Addr>().unwrap_err(),
```

Comment:
You used an incorrect type here, should be `SocketAddrV4`

Reply:
fixed